examples/gnrc_lorawan/Makefile

```diff
@@ -20,36 +20,50 @@ DRIVER ?= sx1276
 
 USEMODULE += $(DRIVER)
 
+include $(RIOTBASE)/Makefile.include
+
 ########################### COMPILE TIME CONFIGURATION ########################
 # NOTE: The following options can be configured on runtime as well using
 # `ifconfig`
 
+# The configurations can be accessed via menuconfig using `make menuconfig`
+
 # OTAA compile time configuration keys
-CFLAGS += -DLORAMAC_APP_KEY_DEFAULT=\{0x00\,0x00\,0x00\,0x00\,0x00\,0x00\,0x00\,0x00\,0x00\,0x00\,0x00\,0x00\,0x00\,0x00\,0x00\,0x00\}
-CFLAGS += -DLORAMAC_APP_EUI_DEFAULT=\{0x00\,0x00\,0x00\,0x00\,0x00\,0x00\,0x00\,0x00\}
-CFLAGS += -DLORAMAC_DEV_EUI_DEFAULT=\{0x00\,0x00\,0x00\,0x00\,0x00\,0x00\,0x00\,0x00\}
+ifeq (,$(CONFIG_LORAMAC_APP_KEY_DEFAULT)$(CONFIG_LORAMAC_APP_EUI_DEFAULT)$(CONFIG_LORAMAC_DEV_EUI_DEFAULT))
+CFLAGS += -DCONFIG_LORAMAC_APP_KEY_DEFAULT=\"AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA\"
+CFLAGS += -DCONFIG_LORAMAC_APP_EUI_DEFAULT=\"BBBBBBBBBBBBBBBB\"
+CFLAGS += -DCONFIG_LORAMAC_DEV_EUI_DEFAULT=\"CCCCCCCCCCCCCCCC\"
+endif
 
 # Uncomment and replace with proper keys for joining with ABP
 # For TTN, It's necessary to set the RX2 DR to 3
 # NOTE: This values will be overriten in case of OTAA.
-#CFLAGS += -DLORAMAC_DEV_ADDR_DEFAULT=\{0x00\,0x00\,0x00\,0x00\}
-#CFLAGS += -DLORAMAC_NWK_SKEY_DEFAULT=\{0x00\,0x00\,0x00\,0x00\,0x00\,0x00\,0x00\,0x00\,0x00\,0x00\,0x00\,0x00\,0x00\,0x00\,0x00\,0x00\}
-#CFLAGS += -DLORAMAC_APP_SKEY_DEFAULT=\{0x00\,0x00\,0x00\,0x00\,0x00\,0x00\,0x00\,0x00\,0x00\,0x00\,0x00\,0x00\,0x00\,0x00\,0x00\,0x00\}
-#CFLAGS += -DLORAMAC_DEFAULT_RX2_DR=LORAMAC_DR_3
+# ifeq (,$(CONFIG_LORAMAC_DEV_ADDR_DEFAULT)$(DCONFIG_LORAMAC_NWK_SKEY_DEFAULT)$(DCONFIG_LORAMAC_APP_SKEY_DEFAULT))
+# CFLAGS += -DCONFIG_LORAMAC_DEV_ADDR_DEFAULT=\"00000000\"
+# CFLAGS += -DCONFIG_LORAMAC_NWK_SKEY_DEFAULT=\"00000000000000000000000000000000\"
+# CFLAGS += -DCONFIG_LORAMAC_APP_SKEY_DEFAULT=\"00000000000000000000000000000000\"
+# endif
+
+## For FIT-IoT Lab usage. Use the highest DR since gateway is nearby.
+ifndef CONFIG_LORAMAC_DEFAULT_RX2_DR
+CFLAGS += -DCONFIG_LORAMAC_DEFAULT_DR_5
+endif
```

Comment:
For Applications in FIT IoT Lab, higher DR is preferred while for outdoor uses lower DR is preferred at least to start with.

Reply:
What's the relationship between the RX2 DR and the fact is running on IoT-LAB?

Also, it's not 100% accurate IMO that the lower DR is preferred in outdoor applications. Strictly speaking this is application specific (there are applications e.g mobile objects that require a higher throughput and the nodes are not too far away). I think I wouldn't give it too much importance to the default value, I would rather try to give visibility to these configuration parameters in case someone wants to use default values.

Reply:
Since most of my testing initially was around FIT IoT lab, higher DR was good to have as the default. Lower DR was not always reliable for some reason. I do agree that it depends more on the use case but for long-range/outdoor applications, lower DR fares better than higher DR. 

I have commented the same in Makefile